### PR TITLE
UpdateExpressions: add necessary changes to tests to reflect latest specification.

### PR DESCRIPTION
**Pull request by rwaldron**

### CODE REVIEW

File: test/language/expressions/postfix-decrement/arguments.js [hunk 1]
```diff
@@ -2,11 +2,21 @@
 // This code is governed by the BSD license found in the LICENSE file.
 
 /*---
-es5id: 11.3.2-2-1-s
-esid: sec-postfix-decrement-operator
+esid: sec-update-expressions
 description: >
-    Strict Mode - SyntaxError is thrown if the identifier 'arguments'
-    appear as a PostfixExpression(arguments--)
+  It is an early Syntax Error if AssignmentTargetType of LeftHandSideExpression is strict. (arguments)
```
Comment (leobalter) [inline test/language/expressions/postfix-decrement/arguments.js]: not sure if this should be unified to `test/language/expressions/postfix-decrement/arguments-nostrict.js`, are the semantics there only applying to no strict?

Comment (leobalter) [reply test/language/expressions/postfix-decrement/arguments.js]: same for `test/language/expressions/postfix-decrement/eval.js`

Comment (leobalter) [reply test/language/expressions/postfix-decrement/arguments.js]: and the other postfix/prefix-increment equivalents

Comment (rwaldron) [reply test/language/expressions/postfix-decrement/arguments.js]: Did you see my last comment here: https://github.com/tc39/test262/pull/1642#issuecomment-448316186? There are explicit semantics for strict mode and non-strict mode, and those semantics are different.

Comment (leobalter) [reply test/language/expressions/postfix-decrement/arguments.js]: Thanks for the clarification, I missed that notification.

**Comment by adrianheine** [comment]
I can't quite follow the `-nostrict` tests.

[sec-identifiers-static-semantics-assignmenttargettype](https://tc39.github.io/ecma262/#sec-identifiers-static-semantics-assignmenttargettype):

>  IdentifierReference : Identifier
> 
> 1. If this IdentifierReference is contained in strict mode code and StringValue of Identifier is "eval" or "arguments", return strict.
> 2. Return simple. 

So `AssignmentTargetType` of `arguments` in non-strict mode code is `simple`.

[sec-update-expressions-static-semantics-early-errors](https://tc39.github.io/ecma262/#sec-update-expressions-static-semantics-early-errors):

> 12.4.1 Static Semantics: Early Errors
> UpdateExpression:
> LeftHandSideExpression++
> LeftHandSideExpression--
> 
>  It is an early Reference Error if AssignmentTargetType of LeftHandSideExpression is invalid.
>  It is an early Syntax Error if AssignmentTargetType of LeftHandSideExpression is strict.

`AssignmentTargetType` of `LeftHandSideExpression` (`arguments`) is `simple`, so this does not evaluate to an early error.

> (cont.)
>
> UpdateExpression:
> ++UnaryExpression
> --UnaryExpression
> 
>  It is an early Reference Error if AssignmentTargetType of UnaryExpression is invalid.
>  It is an early Syntax Error if AssignmentTargetType of UnaryExpression is strict.

`AssignmentTargetType` of `UnaryExpression` (`arguments`) is `simple`, so this does not evaluate to an early error.

The tests quote [sec-update-expressions-static-semantics-assignmenttargettype](https://tc39.github.io/ecma262/#sec-update-expressions-static-semantics-assignmenttargettype), but that defines the `AssignmentTargetType` of `UpdateExpression`. `AssignmentTargetType` of `arguments--` is `invalid`, but it is not tested anywhere in the `ExpressionStatement` `arguments--;`

**Pull request merged**